### PR TITLE
GIP-0066 and GIP-0068

### PR DESCRIPTION
**Pull request by tmigone**

### CODE REVIEW

File: gips/0066-graph-horizon.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+# Motivation
+
+The Graph launched over two years ago with the objective of being an efficient and censorship-resistant protocol that can scale to organize all of the world’s public data and information. Since then there have been significant changes in the ecosystem and we’ve learned from conversations with many users and contributors. [GIP-0042](https://forum.thegraph.com/t/gip-0042-a-world-of-data-services/3761) introduced the concept of data services, recognizing that other approaches besides subgraphs are possible. [Graph Horizon](https://forum.thegraph.com/t/graph-horizon-explained-a-proposal-for-the-evolution-of-the-protocol/5169) expands on this idea, presenting a modular, composable and permissionless staking and payments protocol that can be leveraged by different data service implementations to offer a wide range of solutions to satisfy all data needs.
```
Comment (PedroMD) [inline gips/0066-graph-horizon.md:25]: > The Graph launched over two years ago

If we're referring to the network (mainnet), shouldn't we say 4 years, or reference the actual date that is December 2020?

File: gips/0066-graph-horizon.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+As described before, the slashing risk applies both to stake provisioned by the service provider and delegated stake. To mitigate the risk of service providers misappropriating delegator funds by using a malicious data service via slashing, we introduce a few protections for delegators. First, the staking contract ensures that all ****service provider stake will be slashed before the delegator’s stake is touched. Since a portion of the slashed amount is burned it greatly reduces the economic viability of an attack. Moreover, when slashing delegators the tokens can only be burned and not used as slashing reward, this means that a malicious data service cannot profit directly from slashing delegators. In general, the more stake the service provider has on a provision the safer delegators will be. This protections in combination with per data service delegation make delegators safe from these types of attacks.
```
Comment (PedroMD) [inline gips/0066-graph-horizon.md:107]: >****service provider 

markdown formatting error?

File: gips/0066-graph-horizon.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+This example might lead the reader into the false conclusion that delegating to the service provider with the highest amount of stake is the safest option. It’s worth noting that in the example we assumed the slashed amount is the same for both cases. In a real world scenario it’s up to each data service to establish their slashing rules that determine what the slashing amount is. The Data Service framework establishes some rules but additional policies might be implemented as well, for instance delegation ratio can be taken into consideration when choosing the slashed amount, or repeated offenses can be punished more severely. Choosing the a service provider to delegate to should be a balancing act that looks at different variables like the service provider’s stake, the delegation ratio, fee cuts for delegation, etc. We expect tools to emerge that will expose this information and provide a simple way for delegators of knowing how aligned service providers and data services are with The Graph standards. Similar tools exist already in the current version of the protocol but they are mostly aimed towards indexer operations (for example, Graphtronauts, Graphscan, Indexer Tools, etc).
```
Comment (PedroMD) [inline gips/0066-graph-horizon.md:113]: > (for example, Graphtronauts, Graphscan, Indexer Tools, etc).

minor comment: GraphSeer could also be listed here

File: gips/0066-graph-horizon.md (hunk 2 above)
Comment (PedroMD) [inline gips/0066-graph-horizon.md:107]: > In general, the more stake the service provider has on a provision the safer delegators will be.

Is this true, considering the slashable amount is defined on a data service basis, as mentioned in the section below? Or is it generally true, considering that some rules are _enforced/(encouraged?)_ by the Data Service Framework?

Comment (PedroMD) [reply gips/0066-graph-horizon.md:107]: The comment above didn't take into account this part found below.

> As a general guideline however, the Data Service framework proposes the slashing amount should be proportional to the provisioned stake at the time a dispute is raised.

So I guess this is generally true if the Data Service follows the Frameworks' guidelines.

File: gips/0066-graph-horizon.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+When the service provider creates a provision there are a few considerations to make. They need to specify the slashing authority for the provision and the thawing period for withdrawing the provisioned stake. While setting these parameters is entirely under the service provider’s control it’s expected that data services will only accept provisions that fulfill their operational requirements (namely, that the data service contract is the slashing authority and that the thawing period is greater than whatever the data service requires).
+
+It’s important to note that, while similar, provisions are not what the current protocol defines as allocations but rather a generalized version that operates at a higher level. The provision primitive guarantees that a certain amount of service provider stake is locked as economic security to provide a service. How this security is used is up to the data service implementation, the staking protocol offers a set of interfaces to perform common operations like thawing provisioned stake, managing stake delegations, etc.
+
+![provisions](../assets/gip-0066/provisions.png)
```
Comment (PedroMD) [inline gips/0066-graph-horizon.md:66]: These diagrams were exported without a background, which makes them hard to read if rendered in dark backgrounds (like here on Github if using the dark theme). Maybe consider updating with a white background like some other diagrams found in other sections of this GIP?

**Review by pcarranzav — APPROVED** [review]
These have been published now so I suggest merging and addressing @p-diogo's feedback in new PRs

**Comment by RembrandtK** [comment]
I see approved, but not closed, and not merged onto main. We are ready to merge and close this PR?

**Comment by tmigone** [comment]
Yes @RembrandtK, I'll merge now.

**Comment by tmigone** [comment]
Actually I can't merge as I don't have write access, @p-diogo can you help here?

**Pull request merged**